trust-quorum/src/node.rs

```diff
@@ -76,15 +76,70 @@ impl Node {
         Ok(persistent_state)
     }
 
+    /// Process a timer tick
+    ///
+    /// Ticks are issued by the caller in order to move the protocol forward.
+    /// The current time is passed in to make the calls deterministic.
+    pub fn tick(&mut self, now: Instant, outbox: &mut Vec<Envelope>) {
+        self.send_coordinator_msgs(now, outbox);
+    }
+
+    /// Handle a message from a another node
+    pub fn handle(
+        &mut self,
+        _now: Instant,
+        _outbox: &mut Vec<Envelope>,
+        from: PlatformId,
+        msg: PeerMsg,
+    ) -> Option<PersistentState> {
+        match msg {
+            PeerMsg::PrepareAck(epoch) => {
+                self.handle_prepare_ack(from, epoch);
+                None
+            }
+            _ => todo!(),
+        }
+    }
+
+    /// Return the current state of the coordinator
+    pub fn get_coordinator_state(&self) -> Option<&CoordinatorState> {
+        self.coordinator_state.as_ref()
+    }
+
+    fn handle_prepare_ack(&mut self, from: PlatformId, epoch: Epoch) {
+        // Are we coordinating for this epoch?
+        if let Some(cs) = &mut self.coordinator_state {
+            let current_epoch = cs.reconfigure_msg().epoch();
+            if current_epoch == epoch {
+                // Store the ack in the coordinator state
+                cs.ack_prepare(from);
+            } else {
+                // Log and drop message
```

Comment:
should this `warn` and the one below be `debug_assert`?

Reply:
We don't want to `debug_assert` here. This could just be a delayed response, once coordination had completed. The prepare phase of reconfiguration doesn't necessarily wait for a response from all nodes, and Nexus may have already committed. Nexus may also have started a new configuration at another node, which would implicitly cancel this reconfiguration when it got a prepare with a higher epoch from the new coordinator. Implementing that behavior is coming soon!